src/Tizen.NUI/src/internal/Application.cs

```diff
@@ -605,32 +605,26 @@ public enum MemoryStatus
         {
             add
             {
-                lock (this)
+                // Restricted to only one listener
+                if (_applicationInitEventHandler == null)
```

Comment:
I agree to remove `lock(this)` in event pattern. but I just want to point out the name of handlers. 
after this patch is merged, could you upload new PR to remove these `_`  prefix?
in c#, pascal case is supposed to be used and for private member, it would be OK to use `_` prefix, but in NUI, we have decided not to use prefix such as `_`, `m`, `p`, and etc.

- NUI ground rule : https://code.sec.samsung.net/confluence/display/GFX/NUI+Ground+Rule
- NUI coding rule : https://code.sec.samsung.net/confluence/pages/viewpage.action?pageId=191373542

Reply:
Okay, I will do it in new PR.